scripts/echoserver/main.go

```diff
@@ -1,18 +1,33 @@
 package main
 
-// A simple echo server.  It listens on a random port, prints that port, then
-// echos back anything sent to it.
+// A simple echo server that listens on the specified network (tcp4 or tcp6) and
+// port, prints the resulting port (since you can use 0 to get a random port),
+// then echos back anything sent to it.  This is to test counting applications
+// that use port forwarding; currently only JetBrains uses this method.
+// Example usage: go run ./scripts/echoserver tcp6 0 -Didea.vendor.name=JetBrains
 
 import (
 	"errors"
 	"fmt"
 	"io"
 	"log"
 	"net"
+	"os"
 )
 
 func main() {
-	l, err := net.Listen("tcp", "127.0.0.1:0")
+	network := os.Args[1]
+	var address string
+	switch network {
```

Comment:
as mentioned above, we should always listen on `"tcp", "[::1]:0"` which gives you a dual stack listener on the same port.